src/daft-csv/src/read.rs

```diff
@@ -371,11 +339,19 @@ where
             yield chunk_buffer
         }
     };
+    Box::pin(read_stream)
+}
+
+#[allow(clippy::too_many_arguments)]
+fn parse_into_column_array_chunk_stream(
```

Comment:
it may be better performance to have a generic of `ByteRecordChunkStream` rather than do `Box<dyn ByteRecordChunkStream>`

Reply:
Already tried that and ran into a bunch of issues, but I'll take another crack at it.